main.go

```diff
@@ -281,32 +281,34 @@ func promptForProfile(p []string) (string, error) {
 	return result, nil
 }
 
-func validateInstance(t TerraformInstanceResult) ResourceInstance {
-	if t.Error != nil {
-		fmt.Printf("Something went wrong getting the terraform state.\n")
-		fmt.Println(t.Error)
-		os.Exit(1)
-	}
-
-	instanceMap := t.Instances
+func validateInstance(t map[string]resourceInstance) resourceInstance {
 
 	if resource == "" {
-		resource = promptForInstance(instanceMap)
+		resource = promptForInstance(t)
 	}
 
-	if !resourceInState(resource, instanceMap) {
-		fmt.Printf("Could not find resource %s. Instances available: %v\n", resource, instanceMap)
+	if !resourceInState(resource, t) {
+		fmt.Printf("Could not find resource %s. Instances available: %v\n", resource, t)
 	}
 
-	return instanceMap[resource]
+	return t[resource]
 }
 
 func main() {
 	flag.Parse()
 
-	// run terraform to get the state
-	c := make(chan TerraformInstanceResult)
-	go getInstances(c)
+	c := make(chan map[string]resourceInstance)
+	go func(c chan map[string]resourceInstance) {
+		// run terraform to get the state
+		fmt.Printf("Getting terraform state...")
```

Comment:
Since you are exiting on error, this doesn't matter so much, but.... here's a desc of closing channels cleanly  https://go101.org/article/channel-closing.html